### ISSUE TITLE
--no-default-features support

### ISSUE THREAD

**Issue by o0Ignition0o** [rust-lang/rust-clippy#4860]
As explained [in our attempts to progressively enable clippy::pedantic to neqo](https://github.com/mozilla/neqo/issues/150), we ended up trying to feature-gate `#[deny(warnings)]`, having it enabled as the default feature set.

As I'm trying to work on [cargo-scout](https://github.com/o0Ignition0o/cargo-scout), which will only warn you on lints that affect your diff / pull request, I'm trying to run clippy with the default feature set disabled (and enabling features if / when required).

I have noticed [there's a way to tell cargo to not compile default features and override them](https://doc.rust-lang.org/cargo/reference/manifest.html#usage-in-end-products) by using the `--no-default-features` flag

Maybe I have misread the documentation or misused the options, but I can't seem to be able to propagate it to clippy.

I'm willing to work on a way to pass this flag to clippy, as well as maybe handle `--features "foo bar baz"` in the process, but I have never worked on clippy yet.

The command would look something like:
```
cargo clippy --features "deny-warnings" -- -W clippy::pedantic
cargo clippy --no-default-features -- -W clippy::pedantic
```
I suppose it could be achieved, given clippy already has an `all-features` flag, but I didn't dig too far yet.

Could someone please let me know if that would make sense, if it was ok if I tried to work on it, and if someone could mentor me / provide me with general directions as to how to proceed ?

Thank you again for your time :)

**Comment by flip1995** [comment]
I just tested this and Clippy, worked with both `--features` and `--no-default-features`. All the `features` and `target` flags should just get passed through to cargo. So if it works with `cargo check` it should work with `cargo clippy`.

**Comment by o0Ignition0o** [comment]
Oh ok, I'm going to give it an other try in an hour, maybe there's a bug somewhere else then... thank you for your time

**Comment by flip1995** [comment]
Please let us know if you get different results with `cargo check` than with `cargo clippy`. Note that the `features` flags won't get passed to the members of workspaces.

**Comment by o0Ignition0o** [comment]
Good catch, this is the exact setup I'm working with.

I'm going to give it an other try in a workspace subfolder then, which will probably turn out to be ok, and i will then dig through https://github.com/rust-lang/cargo/issues/4942 to work around it, thanks a lot for the hint !

**Comment by o0Ignition0o** [comment]
I just tested it and you were correct, the command works fine in a workspace subfolder.
Closing the issue, thank's a lot for your time and really great insights !
